### PR TITLE
client/web: move api handler into web.go

### PR DESCRIPTION
**Pull request by soniaappasamy**
Also uses `http.HandlerFunc` to pass the handler into `csrfProtect` so we can get rid of the extraneous `api` struct.

### CODE REVIEW

File: client/web/web.go [hunk 1]
```diff
@@ -280,6 +293,15 @@ func Handle(w http.ResponseWriter, r *http.Request) {
...
 	}
+	switch {
+	case strings.HasPrefix(r.URL.Path, "/localapi/"):
+		if c.APIHander == nil {
+			w.WriteHeader(http.StatusInternalServerError)
+			json.NewEncoder(w).Encode("missing api handler")
+			return
+		}
+		c.APIHander.proxyRequestToLocalAPI(w, r)
+	}
```
Comment (soniaappasamy) [inline client/web/web.go:304]: May not want this piece quite yet.

File: client/web/api.go [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+import (
+	"io"
+	"net/http"
+	"strings"
+
+	"tailscale.com/client/tailscale"
+	"tailscale.com/client/tailscale/apitype"
+)
+
+type apiHandler struct {
```
Comment (willnorris) [inline client/web/api.go:15]: do we need the extra apiHandler type?  What if we just put the *tailscale.LocalClient directly on the main WebClient type (or whatever we call it)?

Comment (soniaappasamy) [reply client/web/api.go:15]: yeah fine with that, probably don't need the extra layer of abstraction

File: client/web/web.go [hunk 3]
```diff
@@ -280,6 +293,15 @@ func Handle(w http.ResponseWriter, r *http.Request) {
 		io.WriteString(w, authenticationRedirectHTML)
 		return
 	}
+	switch {
+	case strings.HasPrefix(r.URL.Path, "/localapi/"):
+		if c.APIHander == nil {
```
Comment (willnorris) [inline client/web/web.go:298]: if we remove the apihandler type, then even a zero value tailscale.LocalClient should function, so I don't think we'd need any special error handling here?

File: client/web/api.go [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+func newAPIHandler() *apiHandler {
+	return &apiHandler{lc: &localClient}
+}
+
+// proxyRequestToLocalAPI proxies the web API request to the localapi.
+//
+// The web API request path is expected to exactly match a localapi path,
+// with prefix /localapi/.
+func (h *apiHandler) proxyRequestToLocalAPI(w http.ResponseWriter, r *http.Request) {
+	if !strings.HasPrefix(r.URL.Path, "/localapi/") {
```
Comment (willnorris) [inline client/web/api.go:28]: big TODO here about securing this endpoint before merging this 😄

**Comment by soniaappasamy** [comment]
Repurposed this PR to _just_ include the initial move of api.go into web.go. Will follow-up with the localapi proxy changes.

**Review by willnorris — APPROVED** [review]

**Pull request merged**